src/gmres_e.m

```diff
@@ -0,0 +1,426 @@
+function [x, flag, relresvec, time] = ...
+    gmres_e(A, b, m, k, tol, maxit, xInitial, varargin)
+    % GMRES-E algorithm
+    %
+    %   GMRES-E is a modified implementation of the restarted
+    %   Generalized Minimal Residual Error or GMRES(m) [1], performed by
+    %   appending 'k' eigenvectors corresponding to a few of the smallest
+    %   eigenvalues in magnitude for each outer iteration. In practice,
+    %   the approximate eigenvectors are the harmonic Ritz vectors
+    %   associated to the harmonic Ritz values per outer iteration.
+    %
+    %   Signature:
+    %   ----------
+    %
+    %   [x, flag, relresvec, time] = ...
+    %       lgmres(A, b, m, k, tol, maxit, xInitial)
+    %
+    %
+    %   Input Parameters:
+    %   -----------------
+    %
+    %   A:          n-by-n matrix
+    %               Left-hand side of the linear system Ax = b.
+    %
+    %   b:          n-by-1 vector
+    %               Right-hand side of the linear system Ax = b.
+    %
+    %   m:          int
+    %               Restart parameter (similar to 'restart' in MATLAB).
+    %
+    %   k:          int
+    %               Number of eigenvectors corresponding to
+    %               a few of the smallest eigenvalues in magnitude\
+    %               for each outer iteration.
+    %
+    %   tol:        float, optional
+    %               Tolerance error threshold for the relative residual norm.
+    %               Default is 1e-6.
+    %
+    %   maxit:      int, optional
```

Comment:
What is the default value if not provided?

Reply:
If not provided, it's $min{n, 10}$. Comment should be included in _all_ docstrings.